scripts/samplesheet_generator.py

```diff
@@ -149,6 +146,57 @@ def gen_Hiseq_lane_data(pro):
 
     return ("{}{}".format(header, str_data), data)
 
+
+def gen_Novaseq_lane_data(pro):
+    data=[]
+    header_ar = ["FCID","Lane","Sample_ID","Sample_Name","Sample_Ref","index","index2","Description","Control","Recipe","Operator","Sample_Project"]
+    for out in pro.all_outputs():
+        if  out.type != "Analyte":
+            continue
+        for sample in out.samples:
+            sp_obj = {}
+            sp_obj['lane'] = out.location[1].split(':')[0].replace(',','')
+            sp_obj['sid'] = "Sample_{}".format(sample.name).replace(',','')
+            sp_obj['sn'] = sample.name.replace(',','')
+            try:
+                sp_obj['pj'] = sample.project.name.replace('.','__').replace(',','')
+            except:
+                #control samples have no project
+                continue
+            try:
+                if pro.udf.get('Read 2 Cycles'):
+                    if str(pro.udf['Read 2 Cycles']).replace(',','')==str(pro.udf['Read 1 Cycles']).replace(',',''):
+                        sp_obj['rc'] = "2x{}".format(str(pro.udf['Read 1 Cycles']).replace(',',''))
+                    else:
+                        sp_obj['rc'] = "{}-{}".format(str(pro.udf['Read 1 Cycles']).replace(',',''),str(pro.udf['Read 2 Cycles']).replace(',',''))
+                else:
+                    sp_obj['rc'] = "1x{}".format(str(pro.udf['Read 1 Cycles']).replace(',',''))
+            except:
+                sp_obj['rc'] = ''
+            sp_obj['ct'] = 'N'
+            sp_obj['op'] = pro.technician.name.replace(" ","_").replace(',','')
+            sp_obj['fc'] = out.location[0].name.replace(',','')
+            sp_obj['sw'] = out.location[1].replace(',','')
+            try:
+                sp_obj['ref'] = sample.project.udf['Reference genome'].replace(',','')
+            except:
+                sp_obj['ref']=''
+            if 'use NoIndex' in pro.udf and pro.udf['use NoIndex'] == True:
+                sp_obj['idx1'] = "NoIndex"
+            else:
+                idxs = find_barcode(sample, pro)
+                sp_obj['idx1'] = idxs[0].replace(',','')
+                if idxs[1]:
+                    sp_obj['idx2'] = idxs[1].replace(',','')
+            data.append(sp_obj)
+    header = "{}\n".format(",".join(header_ar))
+    str_data = ""
+    for line in sorted(data, key=lambda x: x['lane']):
+        l_data = [line['fc'], line['lane'], line['sn'], line['sn'], line['ref'], line['idx1'], line['idx2'], line['pj'], line['ct'], line['rc'], line['op'], line['pj']]
```

Comment:
Look like here you always expect `line['idx2']`, but if it is a single index the `if` in line 189 will not pass and the `idx2` will not be defined right ?